builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -92,8 +98,9 @@ func (is *Source) resolveLocal(refStr string) (*image.Image, error) {
 	return img, nil
 }
 
-func (is *Source) resolveRemote(ctx context.Context, ref string, platform *ocispec.Platform, sm *session.Manager, g session.Group) (digest.Digest, []byte, error) {
+func (is *Source) resolveRemote(ctx context.Context, ref string, platform *ocispec.Platform, sm *session.Manager, g session.Group) (string, digest.Digest, []byte, error) {
```

Comment:
Looks like this commit is needed for the BuildKit update (otherwise it doesn't compile)

Reply:
I'd rather keep this separate, but it is possible to first stub this (like I did the first time), and then apply Tonis's implementation on top. That's not my preference, however, as there will have to be non-buildable commits in this PR to keep things reviewable/discrete.